i18n/i18nTextCollector.php

```diff
@@ -44,98 +52,273 @@ class i18nTextCollector extends Object {
 	protected $writer;
 	
 	/**
+	 * List of file extensions to parse
+	 *
+	 * @var array
+	 */
+	protected $fileExtensions = array('php', 'ss');
+	
+	/**
 	 * @param $locale
 	 */
 	public function __construct($locale = null) {
-		$this->defaultLocale = ($locale) ? $locale : i18n::get_lang_from_locale(i18n::default_locale());
+		$this->defaultLocale = $locale
+			? $locale
+			: i18n::get_lang_from_locale(i18n::default_locale());
 		$this->basePath = Director::baseFolder();
 		$this->baseSavePath = Director::baseFolder();
 		
 		parent::__construct();
 	}
 
+	/**
+	 * Assign a writer
+	 * 
+	 * @param i18nTextCollector_Writer $writer
+	 */
 	public function setWriter($writer) {
 		$this->writer = $writer;
 	}
 
+	/**
+	 * Gets the currently assigned writer, or the default if none is specified.
+	 * 
+	 * @return i18nTextCollector_Writer
+	 */
 	public function getWriter() {
-		if(!$this->writer) $this->writer = new i18nTextCollector_Writer_RailsYaml();
+		if(!$this->writer) {
+			$this->writer = Injector::inst()->get('i18nTextCollector_Writer');
```

Comment:
Should this not use `setWriter` - if ever there is any logic that needs to be attached whenever a writer is set, it should go in the setWriter, so that should be the only function to ever do this assignment